src/application/middlewares/auth.ts

```diff
@@ -40,6 +40,9 @@ export class AuthMiddleware implements Middleware {
       if (!payload) {
         return unauthorized(new InvalidTokenError('Invalid or expired token'))
       }
+      if (!payload.sub) {
+        return unauthorized(new InvalidTokenError('Invalid or expired token'))
+      }
 
       const { sub: userId } = payload
```

Comment:
This bug really make sense in this case, because is not expected an empty payload in use cases. But there is not problem in an empty payload in general.

Reply:
If the payload is empty "payload = {}" will go inside the first "if". But if there is something inside the payload that doesn't have the "sub" that will be used in the next lines it will go into the second "if". I did it this way to not break the other tests.